src/Impl/RestPublisher.php

```diff
@@ -47,12 +48,29 @@ private function convertToPayloadTypeName($channelSchemaName): string
 
     public function send(Message $message): SuccessResult
     {
-        $json = $this->jsonProvider->getJson($message, $this->payloadTypeName);
-        $actualHeaders = $message->action == Message::PUBLISH_ACTION
-            ? array_merge($this->headers, ['Content-Type' => 'application/json; charset=UTF-8'])
-            : $this->headers;
+        $messageStatus = $this->sendMulti([$message])[0];
```

Comment:
Note the single breaking change in this method:
 - now we always add the content-type header

But it's good. We send the message as json, even if the nested payload field is null (for unpublishing)